integration/query_scheduler_test.go

```diff
@@ -86,9 +86,12 @@ func runTestQuerySchedulerWithMaxUsedInstances(t *testing.T, seriesName string,
 	inUseScheduler := schedulers[0]
 	notInUseScheduler := schedulers[1]
 
-	// We expect the querier to open 4 connections to the in-use scheduler, and 1 connection to the not-in-use one.
-	require.NoError(t, inUseScheduler.WaitSumMetricsWithOptions(e2e.Equals(4), []string{"cortex_query_scheduler_connected_querier_clients"}))
-	require.NoError(t, notInUseScheduler.WaitSumMetricsWithOptions(e2e.Equals(1), []string{"cortex_query_scheduler_connected_querier_clients"}))
+	// The minimum number of connections per scheduler is 4 in order to avoid queue starvation
+	// when the RequestQueue utilizes the querier-worker queue prioritization algorithm.
+	// Although the max-concurrent is set to 8, the querier will create an extra 4 connections
+	// per not-in-use scheduler meet the minimum requirements per connected RequestQueue instance.
```

Comment:
I think this is missing something to make it make sense?
```suggestion
	// per not-in-use scheduler to meet the minimum requirements per connected RequestQueue instance.
```

Reply:
yep

Reply:
fixed again, sans the merge commit